src/main/java/org/truffleruby/parser/BodyTranslator.java

```diff
@@ -703,21 +705,30 @@ protected ArgumentsAndBlockTranslation translateArgumentsAndBlock(SourceIndexLen
 
         final List<RubyNode> argumentsTranslated = new ArrayList<>();
         final List<RubyNode> trailing = new ArrayList<>();
+        List<RubyNode> stringKeyValues = new ArrayList<>();
```

Comment:
It's not just String keys allowed, any Ruby object is allowed as a key for `**kwrest`.